### PR TITLE
bootstrap: use `nix` crate to get rusage on unix

### PR DESCRIPTION
**Pull request by GrigorenkoPV**
Part of #109859

### CODE REVIEW

File: src/bootstrap/Cargo.toml [hunk 1]
```diff
@@ -47,6 +47,7 @@ fd-lock = "4.0"
 home = "0.5"
 ignore = "0.4"
 libc = "0.2"
+nix = { version = "0.29.0", features = ["resource"] }
```
Comment (klensy) [inline src/bootstrap/Cargo.toml:50]: `cfg(unix)` this?

Comment (GrigorenkoPV) [reply src/bootstrap/Cargo.toml:50]: Oh, yeah, makes sense

**Comment by Kobzol** [comment]
Hi, thanks for the PR! While removing `unsafe` from bootstrap in general is a good idea, this PR adds 4 more dependencies that need to be compiled on every bootstrap rebuild on Linux. I'm not sure if that's a good trade-off, especially when `RUSTC_PRINT_STEP_RUSAGE` seems kind of niche. I don't know about others, but I haven't personally ever used it, and it also doesn't seem to be used on CI.

The dependency cost could be amortized if we used it for more things, of course. I checked https://github.com/rust-lang/rust/issues/109859 and it seems like `nix` could also help with `getuid`, but not with `setpriority` (based on what is currently offered by the `nix` crate). I'm still unsure whether it's a good enough value proposition for `rusage` and `getuid`.

@onur-ozkan What do you think?

**Comment by Kobzol** [comment]
For context, prior discussion: https://github.com/rust-lang/rust/pull/116476

**Comment by onur-ozkan** [comment]
I am not afraid of having `unsafe` blocks in bootstrap, but it's definitely an improvement if we can replace them with safe blocks. However, adding dependencies to achieve this brings more problems than solutions. Also, many of these dependencies follow the logic we already do in bootstrap, which is why we decided not to merge https://github.com/rust-lang/rust/pull/116476.

It seems we should update some items of #109859 as it is leading to incorrect directions that we don't want to follow on bootstrap.

**Comment by GrigorenkoPV** [comment]
> The dependency cost could be amortized if we used it for more things, of course. I checked #109859 and it seems like `nix` could also help with `getuid`, but not with `setpriority` (based on what is currently offered by the `nix` crate). I'm still unsure whether it's a good enough value proposition for `rusage` and `getuid`.

@onur-ozkan, do I get it right that it's a no from you even in this case?

> It seems we should update some items of #109859 as it is leading to incorrect directions that we don't want to follow on bootstrap.

That's what I've been trying to do:
https://github.com/rust-lang/rust/issues/109859#issuecomment-2254271925


In all honesty, I did not really expect this PR get accepted, I mostly wanted a confirmation that `unsafe` for rusage on unix is a non-issue and will not be addressed.

**Comment by JohnCSimon** [comment]
Triage: 

> I did not really expect this PR get accepted

@GrigorenkoPV Maybe close this PR?

**Comment by GrigorenkoPV** [comment]
> @GrigorenkoPV Maybe close this PR?

I would still love to get a reply to my question above first.


> > The dependency cost could be amortized if we used it for more things, of course. I checked #109859 and it seems like `nix` could also help with `getuid`, but not with `setpriority` (based on what is currently offered by the `nix` crate). I'm still unsure whether it's a good enough value proposition for `rusage` and `getuid`.
> 
> @onur-ozkan, do I get it right that it's a no from you even in this case?

**Comment by onur-ozkan** [comment]
> @onur-ozkan, do I get it right that it's a no from you even in this case?

Yeah: https://github.com/rust-lang/rust/blob/f04f6ca36d2439375d20a98be013384afbab0782/src/bootstrap/src/lib.rs#L290-L294

**Pull request closed without merging**